test/core/test/mocked.test.ts

```diff
@@ -169,3 +169,29 @@ describe('mocked function which fails on toReturnWith', () => {
 test('streams', () => {
   expect(exportedStream).toBeDefined()
 })
+
+describe('temporary mock implementation', () => {
+  test('temporary mock implementation works as expected', () => {
+    const mock = vi.fn(() => 1)
+
+    mock.withImplementation(() => 2, () => {
+      expect(mock()).toBe(2)
```

Comment:
if you have expect inside a callback, don't forget to call
`expect.assertions` beforehand